### ISSUE TITLE
[Blog] Pagination focus not well managed

### ISSUE THREAD

**Issue by svinkle** [Shopify/dawn#694]
## Issue summary

When the blog post comment pagination link is clicked, a hash is added to the URL. Focus is meant to shift to the comment heading on load, however the heading does not receive keyboard focus. This forces the user to navigate from the top of the document.

<details>
  <summary>Screenshots</summary>

 <img width="654" alt="Dawn blog post page, comments heading and pagination controls are highlighted." src="https://user-images.githubusercontent.com/1392632/135107153-afd767fa-7810-4752-8131-8525c18d10ed.png">

</details>

#### HTML

```html
<h2 id="Comments-10759274554">12 comments</h2>

<!-- … -->

<a
  href="/blogs/news/blog-title-here?page=2#Comments-10759274554"
  class="pagination__item link"
  aria-label="Page 2"
>
  2
</a>
```

## Steps to reproduce

1. Go to a Dawn theme store with any supported browser
2. Navigate to a blog list page
3. Activate a blog list link with comments
4. With an active screen reader, use the virtual cursor to navigate to the pagination list of links and activate a link

### Expected

- Keyboard focus to shift to the Comments heading.

### Actual

- Keyboard focus remains at the top of the document.

## Recommendation

Add the `tabindex` attribute to the Comments heading, setting its value to `-1`. This will allow the heading element to receive keyboard focus.

#### HTML

```html
<h2 id="Comments-10759274554" tabindex="-1">12 comments</h2>
```

<!-- #### Diff

```diff
- line removed
+ line added
``` -->

## Specifications

- **Browser:** Any
- **Dawn version:** 2.1.0
- **Component:** Blog
- **WCAG Principal:** Operable
- **WCAG SC:** [2.1.1 Keyboard](https://www.w3.org/WAI/WCAG21/Understanding/keyboard.html)
- **Severity:** Medium
- **Effort:** Low
